### ISSUE TITLE
Crashed when use YYWebImage and Performance

### ISSUE THREAD

**Issue by qiuncheng** [firebase/firebase-ios-sdk#1936]
### [READ] Step 1: Are you in the right place?

  Yeah, I'm using in right place.

### [REQUIRED] Step 2: Describe your environment

  * Xcode version: 9.4.1
  * Firebase SDK version: 5.7.0
  * Firebase Component: Performance (Auth, Core, Database, Firestore, Messaging, Storage, etc)
  * Component version: 2.1.1

### [REQUIRED] Step 3: Describe the problem
Crash log is below:
```objc
 [Firebase/Analytics][I-ACS901006] Received SSL challenge for host. Host: https://app-measurement.com/a
2018-10-12 10:29:29.608096+0800 QwikMatchDev[284:58201] *** Assertion failure in -[GULObjectSwizzler swizzle], /Users/yolo/Work/code/TongZhuoGlobal/Pods/GoogleUtilities/GoogleUtilities/ISASwizzler/GULObjectSwizzler.m:146
2018-10-12 10:29:29.615624+0800 QwikMatchDev[284:58201] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'The original class must be the same as the class returned by object_setClass'
*** First throw call stack:
(0x18eb92fe0 0x18d5f4538 0x18eb92eb4 0x18f62a720 0x102209cb0 0x10089aea4 0x10c6e1a10 0x10c6ee5bc 0x10089a9d8 0x10089dabc 0x107458b3c 0x18f6864cc 0x18eb4142c 0x18eb40d9c 0x18eb3e9a8 0x18ea6eda4 0x18f588d74 0x18f5dd6c4 0x107457a0c 0x18f6862d8 0x18dc5568c 0x18dc5559c 0x18dc52cb4)
```

#### Steps to reproduce:
When using YYWebImage and Performance, it will reproduce.

#### Relevant Code:
- code from [YYWebImage](https://github.com/ibireme/YYWebImage/blob/093193c67bfa2409af93e618a48e3208d63106a5/YYWebImage/YYWebImageOperation.m#L375-L382)
    ```objc
    // this code is from YYWebImage: https://github.com/ibireme/YYWebImage/blob/093193c67bfa2409af93e618a48e3208d63106a5/YYWebImage/YYWebImageOperation.m#L375-L382
    if (![self isCancelled]) {
        _connection = [[NSURLConnection alloc] initWithRequest:_request delegate:[_YYWebImageWeakProxy proxyWithTarget:self]];
        if (![_request.URL isFileURL] && (_options & YYWebImageOptionShowNetworkActivity)) {
            [YYWebImageManager incrementNetworkActivityCount];
        }
    }
    ```

**Comment by paulb777** [comment]
Thanks for the report. Tracking internally at b/117644780

**Comment by jazz-mobility** [comment]
Any updates on this. I am getting same crash on 5.9.0

**Comment by mberndt92** [comment]
Same crash, version is 5.9.0. Its crashing inside the class `GULObjectSwizzler`, lines 144-146. 
<img width="722" alt="screen shot 2018-10-19 at 08 58 54" src="https://user-images.githubusercontent.com/23187781/47202429-5da0eb80-d37d-11e8-8eec-0150c2b9ce82.png">

**Comment by kazuhiro4949** [comment]
I'm getting the same assert on v5.11.0. The variables has a class which initializes NSURLConnection.

**Comment by jazz-mobility** [comment]
@paulb777 Any updates on this. I figured out one thing, this crash is only happens in debug mode.

**Comment by morganchen12** [comment]
Can one of you share a stack trace of the crash? The one in the original issue is unsymbolicated. Thanks!

**Comment by kazuhiro4949** [comment]
@morganchen12 
I figure out what's happened in my case. 
I uses KVO and initialize NSURLConnection in class A.
KVO makes  a subclass of A automatically, named: NSKVONotifying_A. 
It doesn't match class name. So the app hooks the above assert.

For example, the following view controller makes crash in debug mode.

```swift
class ViewController: UIViewController {
    @objc var urlConnection: NSURLConnection?
    override func viewDidLoad() {
        super.viewDidLoad()
        addObserver(self, forKeyPath: #keyPath(urlConnection), options: [.new, .old], context: nil)
    }

    override func viewDidAppear(_ animated: Bool) {
        let request = URLRequest(url: URL(string: "https://google.com")!)
        urlConnection = NSURLConnection(request: request, delegate: self, startImmediately: false)
    }
    
    override func observeValue(forKeyPath keyPath: String?, of object: Any?, change: [NSKeyValueChangeKey : Any]?, context: UnsafeMutableRawPointer?) {
    }
}

extension ViewController: NSURLConnectionDelegate, NSURLConnectionDataDelegate {
}
```

**Comment by kazuhiro4949** [comment]
Could you tell me whether the assertion has a problem or not? 
It is an internal specification of KVO.

**Comment by morganchen12** [comment]
@kazuhiro4949 can you share the stack trace of your repro example's crash? GoogleUtilities shouldn't be conflicting with NSURLConnection.

**Comment by kazuhiro4949** [comment]
@morganchen12 
This is the stack track.

```
 *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'The original class must be the same as the class returned by object_setClass'
*** First throw call stack:
(
	0   CoreFoundation                      0x000000010a06e1bb __exceptionPreprocess + 331
	1   libobjc.A.dylib                     0x000000010960c735 objc_exception_throw + 48
	2   CoreFoundation                      0x000000010a06df42 +[NSException raise:format:arguments:] + 98
	3   Foundation                          0x0000000108ad4877 -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 194
	4   GoogleUtilities                     0x0000000109194bc0 -[GULObjectSwizzler swizzle] + 1312
	5   FirebaseAssert                      0x0000000108123cbd __55-[FPRNSURLConnectionDelegateInstrument registerObject:]_block_invoke + 1231
	6   libdispatch.dylib                   0x000000010b31d602 _dispatch_client_callout + 8
	7   libdispatch.dylib                   0x000000010b32b653 _dispatch_lane_barrier_sync_invoke_and_complete + 132
	8   FirebaseAssert                      0x00000001081237c7 -[FPRNSURLConnectionDelegateInstrument registerObject:] + 116
	9   FirebaseAssert                      0x0000000108126826 __InstrumentInitWithRequestDelegateStartImmediately_block_invoke + 162
	10  FirebaseAssert                      0x00000001080d9f5f $SSo15NSURLConnectionC7request8delegate16startImmediatelyABSg10Foundation10URLRequestV_ypSgSbtcfcTO + 255
	11  FirebaseAssert                      0x00000001080d870d $SSo15NSURLConnectionC7request8delegate16startImmediatelyABSg10Foundation10URLRequestV_ypSgSbtcfC + 61
	12  FirebaseAssert                      0x00000001080d8637 $S14FirebaseAssert14ViewControllerC13viewDidAppearyySbF + 359
	13  FirebaseAssert                      0x00000001080d8753 $S14FirebaseAssert14ViewControllerC13viewDidAppearyySbFTo + 51
	14  UIKitCore                           0x000000010d224544 -[UIViewController _setViewAppearState:isAnimating:] + 947
	15  UIKitCore                           0x000000010d22727d __64-[UIViewController viewDidMoveToWindow:shouldAppearOrDisappear:]_block_invoke + 42
	16  UIKitCore                           0x000000010d2255e2 -[UIViewController _executeAfterAppearanceBlock] + 78
	17  UIKitCore                           0x000000010d84920e _runAfterCACommitDeferredBlocks + 634
	18  UIKitCore                           0x000000010d837b2c _cleanUpAfterCAFlushAndRunDeferredBlocks + 384
	19  UIKitCore                           0x000000010d8582de __34-[UIApplication _firstCommitBlock]_block_invoke_2 + 153
	20  CoreFoundation                      0x0000000109fd362c __CFRUNLOOP_IS_CALLING_OUT_TO_A_BLOCK__ + 12
	21  CoreFoundation                      0x0000000109fd2de0 __CFRunLoopDoBlocks + 336
	22  CoreFoundation                      0x0000000109fcd654 __CFRunLoopRun + 1284
	23  CoreFoundation                      0x0000000109fcce11 CFRunLoopRunSpecific + 625
	24  GraphicsServices                    0x0000000112cfd1dd GSEventRunModal + 62
	25  UIKitCore                           0x000000010d83d81d UIApplicationMain + 140
	26  FirebaseAssert                      0x00000001080dabe7 main + 71
	27  libdyld.dylib                       0x000000010b393575 start + 1
	28  ???                                 0x0000000000000001 0x0 + 1
)
```

**Comment by paulb777** [comment]
Fix planned to go out with GoogleUtilities 5.3.5
